dm.pl

```diff
@@ -171,9 +171,15 @@ sub tar_filelist {
 	find({wanted => sub {
 		return if m#^./DEBIAN#;
 		my $tf = NIC::Archive::Tar::File->new(file=>$_);
+		my @stat = lstat($_);
 		my $mode = (lstat($_))[2] & 07777;
 		$tf->mode($mode);
-		$tf->chown("root", "wheel");
+		if ($UID == 0) {
```

Comment:
is `UID` populated from the environment or from `geteuid` or likewise?

Reply:
https://perldoc.perl.org/perlvar.htm
> $REAL_USER_ID
> $UID
> $<
> The real uid of this process. You can change both the real uid and the effective uid at the same time by using POSIX::setuid() . Since changes to $< require a system call, check $! after a change attempt to detect any possible errors.
> Mnemonic: it's the uid you came from, if you're running setuid.

I did test that it matched up.  That indicates it doesn’t come from ENV

Reply:
I think in that case we want `$EUID`, `EFFECTIVE_USER_ID`. If somebody were crazy enough to run `dm.pl` setuid to get permissions working, `UID` would not work.

Reply:
That would make it think it was root before setuid(0) was called.  If you call setuid(0) it will make you be root.  If you check EUID it will be true even if itMs running as mobile with the setuid bit set somehow.  

Unless I misunderstood something somewhere...

Reply:
https://mattmccutchen.net/suidperl.html was an interesting and relevant read... I don't think we really need to worry about setuid on it... but it looks like $UID was sometimes not set but $< which is supposed to be the same, is...